### DISCUSSION TITLE
Experiencing several seconds page load when cold boot (nextjs)

### DISCUSSION THREAD

**Discussion by mbret** [vercel/vercel#10127] [Help]
### Problem Description

Recently deployed a nextjs app on vercel and we noticed some very long (5 to 8 seconds) page load from time to time. After investigating they seem to be cold boot every time. We are testing this on an empty react page with an empty `getServerSideProps`. We just wanted to assert the speed for a blank page to have our basis.

Cold boot takes approx ~160ms so I figured it should not add that much time.

There are also these lines of logs everytime:

```
- info Loaded env from /var/task/apps/front/.env.local
- info Loaded env from /var/task/apps/front/.env.production
```

Is the app building every time as well ? Is there any way we can reduce these times ?

**Comment by dokocat** [comment]
Is there any response here? We have the same issue and completely removed ssr

**Comment by mbret** [comment]
We now have some page that takes more than 10 secondes. Problem is unfortunately still here even with latest version.

**Comment by leerob** [comment]
Please see this guide for some common suggestions on improving function performance: https://vercel.com/guides/how-can-i-improve-serverless-function-lambda-cold-start-performance-on-vercel

**Comment by mbret** [comment]
our testing page which is completely blank and just display a string does not have any added dependency.

![Screenshot 2023-08-02 at 09 59 35](https://github.com/vercel/vercel/assets/1911240/334cc1ce-e679-4378-b4b3-7678bd48b420)

I am not sure how to optimize dependencies further. Regarding edge function, we could use it on some of the pages but we are using node packages on some other so that is not a solution either.

Is there anything else that could be causing this delays ? Again cold boot is noted as ~160ms on vercel, the rest of the time is just unknown reason.
